debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -446,11 +446,15 @@ protected void generateDeleteRecord(TableId tableId, Object[] oldRowData, boolea
 
     private boolean schemaChanged(List<ReplicationMessage.Column> columns, Table table, boolean metadataInMessage) {
         List<String> columnNames = table.columnNames();
-        int messagesCount = columns.size();
-        if (columnNames.size() != messagesCount) {
+        int tableColumnCount = columnNames.size();
+        int replicationColumnCount = columns.size();
+
+        if (tableColumnCount < replicationColumnCount || (tableColumnCount > replicationColumnCount && (!taskContext.config().skipRefreshSchemaOnMissingToastableData() || untoastedColumnsPresent(table, columns)))) {
```

Comment:
This condition is a little bit convoluted, could it be commented or unwinded into a more straightforward understanding?
Also what will hapen when we have a table that contains a TOASTED column that does not change and a new column will be added. Will this condition detect the situation?

Reply:
I'll split the conditions up, and comment them for clarification.

If there is a new column added, it will show up in the replication message either as null or w/ whatever value it has been updated to. The conditional may *not* catch this, but the code after it will (line 466), so we are safe.

Reply:
@jpechane Done.